### ISSUE TITLE
Download

### ISSUE THREAD

**Issue by PanayotCankov** [NativeScript/nativescript-background-http#2]
The sessions currently support uploads, implement download counterpart.

**Comment by brendanheywood** [comment]
Has anyone got a progress bar for downloads working via this or any other NS plugin?

**Comment by Daxito** [comment]
I have it working, it was a pain but it works.

**Comment by brendanheywood** [comment]
awesome! - got a fork or example?

**Comment by Daxito** [comment]
Well, this is the thing, for some reason, from the progress event itself I was not able to update the UI at all, I mean, it was triggering the event when uploading but the UI was not reflecting the change, thus the progress bar was just doing nothing at all. 
Think of this like in Windows Application for instance, you cannot update the UI from a different threat other than the UI threat, so then you use some sort of delegate in order to update the UI.

In this case, it all sounds weird because in JS there is only one threat, but anyway, I came up with a stupid hacky way of doing this, it worked so then I forgot about finding another way:

Basically the idea is to have a local variable that keeps track of the progress and then there is a timer that monitors this variable and, if something is there, then it uses "pubsub" to publish the event to whatever is interested in, we subscribe to this event somewhere else in order to update the Progress bar and it works.
Code attached, hope it makes sense, otherwise discard:


[webapi.txt](https://github.com/NativeScript/nativescript-background-http/files/796914/webapi.txt)
[handler.txt](https://github.com/NativeScript/nativescript-background-http/files/796929/handler.txt)

**Comment by brendanheywood** [comment]
That code still appears to be an upload progress not down load? I'm after a download progress event

**Comment by Daxito** [comment]
oh wow!!! WTH, I completely misunderstood!

**Comment by LorenDorez** [comment]
Any idea if you guys plan to add this download feature? as http.getFile() keeps crashing on me with OutOfMemory exception and the nativescript-downloadmanager plugin only works for android

**Comment by shivgolani** [comment]
@brendanheywood did you manage to find a solution?

**Comment by brendanheywood** [comment]
I haven't tested it but this looks like a winner:

https://plugins.nativescript.rocks/plugin/nativescript-download-progress

**Comment by shivgolani** [comment]
@brendanheywood this plugin does not support background download.
